scripts/release_scripts/restore_backup.py

```diff
@@ -86,6 +97,17 @@ def check_backup_restoration_status():
         common.run_cmd([GCLOUD_PATH, 'datastore', 'operations', 'list']))
 
 
+def cancel_operation():
+    """Cancels a datastore operation."""
+    python_utils.PRINT('List of operations in progress:\n')
+    check_backup_restoration_status()
+    python_utils.PRINT(
+        'Enter the name of the operation to cancel.')
```

Comment:
This is a bit odd. What exactly should I enter here? Some more guidance might be appropriate.

Also, we should show a warning if cancellation happens because that means the server might be in a corrupt state? Or does no data actually get transferred until the very end?

Reply:
> This is a bit odd. What exactly should I enter here? Some more guidance might be appropriate.

I will need some help on this. We need to check this when doing backup restoration.

From the example [here](https://stackoverflow.com/a/53630367), I think the list operation should print out ids (names) of operation which can be entered by the user for cancellation. But we need to confirm this.

> Also, we should show a warning if cancellation happens because that means the server might be in a corrupt state? Or does no data actually get transferred until the very end?

I think the server is in a corrupt state if we cancel ([Ref](https://cloud.google.com/datastore/docs/export-import-entities#cancel_an_operation)). Should I add a warning message then?

Reply:
Yep, please add the warning and include the link you just sent in the warning too.

For the first part -- I'm not sure exactly what help you need, but feel free to ping me / send me instructions and I'll do it!

Reply:
> Yep, please add the warning and include the link you just sent in the warning too.
Added.

> For the first part -- I'm not sure exactly what help you need, but feel free to ping me / send me instructions and I'll do it!
We will need to run a backup restoration and check what is the output from `check_status` function. Is this possible now?